apps/site/assets/ts/helpers/date.ts

```diff
@@ -77,4 +83,22 @@ export const formatToBostonTime = (
   return formatInTimeZone(dateTime, "America/New_York", formatString);
 };
 
+export const formatRelativeTime = (
+  time: Date,
+  targetDate: Date = new Date()
+): string => {
+  const seconds = differenceInSeconds(time, targetDate);
+  if (seconds <= 30) {
+    return "Arriving";
+  }
+  if (seconds <= 60) {
+    return "Approaching";
```

Comment:
For schedules this should be `<1 minute away`

Reply:
Thanks, fixed.